src/internals/ChartInternal.js

```diff
@@ -35,11 +35,21 @@ export default class ChartInternal {
 	}
 
 	beforeInit() {
+		const $$ = this;
+		const config = $$.config;
+
 		// can do something
+
+		config.onbeforeinit.call($$);
```

Comment:
Whole callback functions are treated and called in this way, but I don't think is invoking unnecessary function call.
So, instead of giving default as function, set `undefined` and then check if `onbeforeinit` is a function and then executed the call

```js
// import isFunction from 'util.js'
isFunction(config.onbeforeinit) && config.onbeforeinit();
```

Reply:
Ah, good idea